op-dispute-mon/mon/resolution/delay.go

```diff
@@ -0,0 +1,53 @@
+package resolution
+
+import (
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/types"
+	monTypes "github.com/ethereum-optimism/optimism/op-dispute-mon/mon/types"
+	"github.com/ethereum-optimism/optimism/op-service/clock"
+)
+
+type DelayMetrics interface {
+	RecordClaimResolutionDelayMax(delay float64)
+}
+
+type DelayCalculator struct {
+	metrics DelayMetrics
+	clock   clock.Clock
+}
+
+func NewDelayCalculator(metrics DelayMetrics, clock clock.Clock) *DelayCalculator {
+	return &DelayCalculator{
+		metrics: metrics,
+		clock:   clock,
+	}
+}
+
+func (d *DelayCalculator) RecordClaimResolutionDelayMax(games []monTypes.EnrichedGameData) {
+	var maxDelay uint64 = 0
+	for _, game := range games {
+		delay := d.getMaxDelay(&game)
+		if delay > maxDelay {
+			maxDelay = delay
+		}
```

Comment:
Can we just use:
```suggestion
		maxDelay := max(d.getMaxDelay(&game), maxDelay)
```

Go types might be an issue, not sure.